packages/core/src/notifications/controller.ts

```diff
@@ -0,0 +1,233 @@
+            return
+        }
+
+        getLogger().verbose('ETAG has changed for notifications category: %s', category)
+
+        this.state[category].payload = JSON.parse(response.content)
+        this.state[category].eTag = response.eTag
+        await this.writeState()
+
+        getLogger().verbose(
+            "Fetched notifications JSON for category '%s' with schema version: %s. There were %d notifications.",
+            category,
+            this.state[category].payload?.schemaVersion,
+            this.state[category].payload?.notifications?.length
+        )
+    }
+
+    private fetchRemotely(category: NotificationType): Promise<ResourceResponse> {
+        const fetcher = new HttpResourceFetcher(category === 'startUp' ? startUpEndpoint : emergencyEndpoint, {
+            showUrl: true,
+        })
+
+        return withRetries(async () => await fetcher.getNewETagContent(this.state[category].eTag), {
+            maxRetries: NotificationsController.retryNumber,
+            delay: NotificationsController.retryIntervalMs,
+            // No exponential backoff - necessary?
+        })
+    }
+
+    /**
+     * Fetch notifications from local files.
+     * Intended development purposes only. In the future, we may support adding notifications
+     * directly to the codebase.
+     */
+    private async fetchLocally(category: NotificationType): Promise<ResourceResponse> {
+        if (!_useLocalFiles) {
+            throw new ToolkitError('fetchLocally: Local file fetching is not enabled.')
+        }
+
+        const uri = category === 'startUp' ? startUpLocalPath : emergencyLocalPath
+        const content = await new FileResourceFetcher(globals.context.asAbsolutePath(uri)).get()
+
+        getLogger().verbose('Fetched notifications locally for category: %s at path: %s', category, uri)
+        return {
+            content,
+            eTag: 'LOCAL_PATH',
+        }
+    }
+
+    /**
+     * Write the latest memory state to global state.
+     */
+    private async writeState() {
+        getLogger().debug('NotificationsController: Updating notifications state at %s', this.storageKey)
+
+        // Clean out anything in 'dismissed' that doesn't exist anymore.
+        const notifications = new Set(
+            [
+                ...(this.state.startUp.payload?.notifications ?? []),
+                ...(this.state.emergency.payload?.notifications ?? []),
+            ].map((n) => n.id)
+        )
+        this.state.dismissed = this.state.dismissed.filter((id) => notifications.has(id))
+
+        await globals.globalState.update(this.storageKey, this.state)
+    }
+
+    static get instance() {
+        if (this.#instance === undefined) {
+            throw new ToolkitError('NotificationsController was accessed before it has been initialized.')
+        }
+
+        return this.#instance
+    }
+}
+
+function registerDismissCommand(extPrefix: string) {
+    const name = `_aws.${extPrefix}.notifications.dismiss`
```

Comment:
nit/non-blocker: because of the "stringly typed" nature of vscode commands, it is helpful if they can be found with a search (grep etc). So for that reason maybe we should explicitly do: 
```suggestion
    const name = isAmazonQ() ? `_aws.amazonq.notifications.dismiss` : '_aws....'
```

not sure though.